.github/workflows/ci-build-macos.yaml

```diff
@@ -45,7 +43,27 @@ jobs:
       - name: Tune Runner VM
         uses: ./.github/actions/tune-runner-vm
 
+      - name: Detect changed files
+        id:   changes
+        uses: apache/pulsar-test-infra/paths-filter@master
+        with:
+          filters: |
```

Comment:
Nit. We could reduce some duplication in these files by writing this filter to its own file and then referencing it by path as described here: https://github.com/apache/pulsar-test-infra/tree/master/paths-filter#advanced-options.

Reply:
good proposal. I wasn't aware of that feature. I'll revisit the PR.

Reply:
Done.